cpp/src/arrow/ipc/adapter.cc

```diff
@@ -179,6 +188,15 @@ Status WriteRowBatch(
   RETURN_NOT_OK(serializer.AssemblePayload());
   return serializer.Write(dst, position, header_offset);
 }
+
+Status GetRowBatchSize(const RowBatch* batch, int64_t* size) {
+  RowBatchWriter serializer(batch);
+  RETURN_NOT_OK(serializer.AssemblePayload());
+  RETURN_NOT_OK(serializer.DataHeaderSize(size));
+  *size += serializer.TotalBytes();
```

Comment:
size is the sum of data header size (set on the previous line) and TotalBytes which is the buffer footprint.  TotalBytes might be better named as TotalBufferBytes because it isn't obvious.